### PR TITLE
Add Azure blob storage provider

### PR DESCRIPTION
**Pull request by avalor1**
Adds a http-challenge provider for Azure blob storage

### CODE REVIEW

File: roles/acme/tasks/challenge/http-01/azbs.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+- name: Install prerequisites on localhost
```
Comment (rndmh3ro) [inline roles/acme/tasks/challenge/http-01/azbs.yml:2]: Is this really necessary? I didn't find anything about it in the docs.

File: docs/http-challenge/azbs.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+| acme_azbs_storage_account_name    | yes      |           | Azure storage account name which should be used
+| acme_azbs_container_name    | yes      |           | Azure container name which will be used/created in Azure storage account
+| acme_azbs_subscription_id | yes       | | Azure subscription id
+| acme_azbs_client_id | yes | | Client ID of service principal/application
+| acme_azbs_secret | yes | | Value of secret of service principal/application (Note: not the ID)
+| acme_azbs_tenant_id | yes | | Tenant ID of service principal/application
+
+## Validation
+
+You have to create a service principal/application in the Azure Active Directory.
```
Comment (rndmh3ro) [inline docs/http-challenge/azbs.md:15]: I don't think that we should document how to create a SP. We should just link to the azure-docs. Maybe to this: https://learn.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals

**Comment by rndmh3ro** [comment]
We should also add the requirement of the azure-collection to the [galaxy.yml](https://github.com/T-Systems-MMS/ansible-collection-acme/blob/master/galaxy.yml):
```
dependencies:
  community.crypto: '>=1.0.0'
  openstack.cloud: '>=1.2.1'
  amazon.aws: '>=5.0.0'
  azure.azcollection: '>=1.14.0'
```

File: docs/http-challenge/azbs.md (hunk 2 above)
Comment (avalor1) [reply docs/http-challenge/azbs.md:15]: You are right. I will just link to the azure-docs

File: roles/acme/tasks/challenge/http-01/azbs.yml (hunk 1 above)
Comment (avalor1) [reply roles/acme/tasks/challenge/http-01/azbs.yml]: True, i removed it.

**Review by rndmh3ro — APPROVED** [review]

**Pull request merged**